python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/scenarios/exotic_partition_mapping_scenarios.py

```diff
@@ -155,6 +162,14 @@
 
 
 exotic_partition_mapping_scenarios = {
+    "self_dependency_never_materialized_many_partitions": AssetReconciliationScenario(
+        assets=one_asset_self_dependency_hourly_many_partitions,
+        unevaluated_runs=[],
+        expected_run_requests=[
+            run_request(asset_keys=["asset1"], partition_key="2015-01-01-00:00")
```

Comment:
This test case reproduces the problem (and the solution) - I am a little confused why this is the expected run request here - wouldn't it be the most recent partition set rather than the earliest one?

Reply:
because it has a self-dependency I think it has to do the first one first. the last one requires the second-to-last one.